### PR TITLE
added linked queue

### PR DESCRIPTION
**Pull request by MichaelBatsian**
Now Tom can use custom linked queue

### CODE REVIEW

File: help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    public class LinkedQueue<T> : IQueue<T>,IEnumerable<T>
+    {
+        private Node<T> _begin;
+        private Node<T> _last;
+        private int _count;
+
+        public int Count => _count;
+
+        public bool IsEmpty()
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs]: Could be used property and arrow function

File: help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        public void Enqueue(T obj)
+        {
+            _count++;
+            if (_begin == null)
+            {
+                _begin = new Node<T>();
+                _begin.Data = obj;
+                _last = _begin;
+                return;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs]: this is not so good to write `return;`

File: help-tom/CustomQueue/Godeltech.Custom/Impl/Node.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿
+namespace Godeltech.Custom.Impl
+{
+    public class Node<T>
+    {
+        public T Data;
+        public Node<T> Next;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/Node.cs]: public field is not so good. Should be public property

File: help-tom/CustomQueue/Godeltech.Custom/Impl/Node.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿
+namespace Godeltech.Custom.Impl
+{
+    public class Node<T>
+    {
+        public T Data;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/Node.cs]: The same about field and property

File: help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
+using System;
+using Godeltech.Custom.Interface;
+
+namespace Godeltech.Custom.Impl
+{
+    public class LinkedQueue<T> : IQueue<T>,IEnumerable<T>
+    {
+        private Node<T> _begin;
+        private Node<T> _last;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs]: `_begin` and `_last` this is not opposite words. begin-end first-last

File: help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
+using System;
+using Godeltech.Custom.Interface;
+
+namespace Godeltech.Custom.Impl
+{
+    public class LinkedQueue<T> : IQueue<T>,IEnumerable<T>
+    {
+        private Node<T> _begin;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs]: better to rename to something more understandable. like topOfTheQueue or something like that.

**Comment by MichaelBatsian** [comment]
Corrected linked queue according to your remarks.

File: help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs (hunk 2 above)
Comment (MichaelBatsian) [reply help-tom/CustomQueue/Godeltech.Custom/Impl/LinkedQueue.cs]: Why "return" bad way here? I thought that is better than "if else"?

File: help-tom/CustomQueue/Godeltech.Applications/Program.cs [hunk 7]
```diff
@@ -12,14 +12,28 @@ static void Main(string[] args)
...
-            string t = queue.Peek();
+            string current = queue.Peek();
             Console.WriteLine(queue.Peek());
             
             for (int i = 0; i < len; i++)
             {
                 Console.WriteLine(queue.Dequeue());
             }
 
+            var  lq = new LinkedQueue<string>();
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Applications/Program.cs]: `lq` - `linkedQueue`

File: help-tom/CustomQueue/Godeltech.Applications/Program.cs [hunk 8]
```diff
@@ -12,14 +12,28 @@ static void Main(string[] args)
...
+            lq.Enqueue("Misha");
+            lq.Enqueue("Sasha");
+            lq.Enqueue("Pasha");
+            lq.Enqueue("Olia");
+            int llen = lq.Count;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Applications/Program.cs]: `llen` - `queueSize`

File: help-tom/CustomQueue/Godeltech.Custom/Impl/Queue.cs [hunk 9]
```diff
@@ -6,55 +6,55 @@ namespace Godeltech.Custom.Impl
     public class Queue<T> : IQueue<T>
     {
         private T[] _array;
-        private int size;
-        private const int defCapacity = 10;
-        private int capacity;
-        private int head;
-        private int tail;
+        private int _size;
+        private const int DefCapacity = 10;
```
Comment (pryabov) [inline help-tom/CustomQueue/Godeltech.Custom/Impl/Queue.cs]: constants before fields

**Pull request merged**